trailblazer/store/crud/delete.py

```diff
@@ -18,7 +18,8 @@ def delete_analysis_jobs(self, analysis: Analysis) -> None:
         """Delete all jobs linked to the given analysis."""
         session: Session = get_session()
         for job in analysis.jobs:
-            session.delete(job)
+            if job.job_type == JobType.ANALYSIS:
+                session.delete(job)
```

Comment:
It is rather ensuring the old behavior, but I get what you mean. I'll add a test to to prevent any bugs being introduced in the future that could cause the upload jobs to be deleted again